### DISCUSSION TITLE
Change hostname after install

### DISCUSSION THREAD

**Discussion by aciescott** [Security-Onion-Solutions/securityonion#2799] [Unsupported Versions]
Creating a base VM for airgap install, So I need to change the hostname on the imported VM. I was able to update the IP address by using so-ip-update

**Comment by TOoSmOotH** [comment]
Changing the host name is not easy. You will need to change a bunch of pillars, all the minions, and delete all the certificates so they can be recreated with the new host name. grep for your host name in the global pillar and fix it in all of those places. Probably easier to just re-install.

**Reply by aciescott** [reply]
I have yet to successfully get Security onion to install without having internet access. Note I also want the Gui so need to do the analyst install as well. So I was hoping I would be able just import a VM and then change the hostname and ip but looks like changing the hostname might be difficult.

**Reply by gcsrellis** [reply]
Install with internet with the appropriate hostname then sneaker net it over to your other network. You'll be able to change the IP easier than the hostname for sure.  I do this for various VM's.

**Reply by defensivedepth** [reply]
@aciescott Are you using a 2nd VM for the analyst workstation?

**Comment by vanh20** [comment]
Yes. It's easier to install. The problems with the certificate also goes when you change the IP Address even when you use so-ip-update. Also my manager (Standalone) becomes "Offline" at the Grid and Fleet web interfaces. Just registering my observations.
